include/quda.h

```diff
@@ -1152,6 +1152,35 @@ extern "C" {
    */
   void dslashQuda(void *h_out, void *h_in, QudaInvertParam *inv_param, QudaParity parity);
 
+  /**
+   * Apply the covariant derivative.
+   * @param h_out  Result spinor field
+   * @param h_in   Input spinor field
+   * @param dir    Direction of application
+   * @param param  Metadata for host and device storage
+   */
+  void covDevQuda(void *h_out, void *h_in, int dir, QudaInvertParam param);
+
+  /**
+   * Apply the covariant derivative.
+   * @param h_out  Result spinor field
+   * @param h_in   Input spinor field
+   * @param dir    Direction of application
+   * @param sym    Apply forward=2, backward=2 or symmetric=3 shift
+   * @param param  Metadata for host and device storage
+   */
+  void shiftQuda(void *h_out, void *h_in, int dir, int sym, QudaInvertParam param);
+
+  /**
+   * Apply the spin-taste operator.
+   * @param h_out  Result spinor field
+   * @param h_in   Input spinor field
+   * @param spin   Spin gamma structure
+   * @param taste  Taste gamma structure
+   * @param param  Metadata for host and device storage
+   */
+  void spinTasteQuda(void *h_out, void *h_in, int spin_, int taste, QudaInvertParam inv_param);
```

Comment:
`spin_` -> `spin` to be consistent with the Doxygen comment

Reply:
Also, instead of making this an int, shouldn't it be of type `QudaSpinTasteGamma`?